### PR TITLE
feat(redshift): expose user.secret as property (#17520)

### LINKED ISSUE #17520: (redshift): expose User.secret as property
### Description

Currently there is no way to access the secret created by Redshift User construct and that make it difficult to use method addRotationMultiUser to rotate the user credentials.
https://github.com/aws/aws-cdk/blob/v1.132.0/packages/@aws-cdk/aws-redshift/lib/user.ts#L150
https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-redshift.User.html

### Use Case

1. The secret needed to allow redshift cluster rotating the credentials, function addRotationMultiUser accept only secret.
2. The secret ARN is needed by other services, for example creating IAM Policy and need to scope down the resources only to this secret.

### Proposed Solution

The solution to add Property called secret that will return the secret created https://github.com/aws/aws-cdk/blob/v1.132.0/packages/@aws-cdk/aws-redshift/lib/user.ts#L150

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by Ahmed-Hussein93**
This change will expose Redshift User.secret as property and close #17520
----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/package.json [hunk 1]
```diff
@@ -534,6 +534,7 @@
     "./aws-medialive": "./aws-medialive/index.js",
     "./aws-mediapackage": "./aws-mediapackage/index.js",
     "./aws-mediastore": "./aws-mediastore/index.js",
+    "./aws-mediatailor": "./aws-mediatailor/index.js",
```
Comment (TheRealAmazonKendra) [inline packages/aws-cdk-lib/package.json:537]: Why is this being added here?

File: packages/@aws-cdk/aws-redshift/lib/user.ts [hunk 2]
```diff
@@ -137,6 +138,12 @@ export class User extends UserBase {
   readonly databaseName: string;
   protected databaseProps: DatabaseOptions;
 
+  /**
+   * the Secrets Manager secret of the user.
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-redshift/lib/user.ts:142]:
```suggestion
   * The Secrets Manager secret of the user.
```

Comment (Ahmed-Hussein93) [reply packages/@aws-cdk/aws-redshift/lib/user.ts]: Fixed

File: packages/aws-cdk-lib/package.json (hunk 1 above)
Comment (Ahmed-Hussein93) [reply packages/aws-cdk-lib/package.json:537]: The build will fail without adding this line.

Comment (alexdennis) [reply packages/aws-cdk-lib/package.json:537]: Curious, how is this related to your change? How did you discover it was missing?

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**